docs/adr/0017-python-anlaytics-processing.md

```diff
@@ -0,0 +1,205 @@
+* Deploy Data Analytics Model Container cloud.gov container registry with CircleCI
+
+
+```yaml
+...
+cf_dam_deploy:
+    description: "Login to cloud foundry space with service account credentials
+      and push application using deployment configuration file."
+    parameters:
+      app_name:
+        description: "Name of Cloud Foundry cloud.gov application; must match
+          application name specified in manifest"
+        type: string
+      auth_client_id:
+        description: "Name of CircleCi project environment variable that
+          holds authentication client id, a required application variable"
+        type: env_var_name
+      auth_client_secret:
+        description: "Name of CircleCi project environment variable that
+          holds authentication client secret, a required application variable"
+        type: env_var_name
+      cloudgov_username:
+        description: "Name of CircleCi project environment variable that
+          holds deployer username for cloudgov space"
+        type: env_var_name
+      cloudgov_password:
+        description: "Name of CircleCi project environment variable that
+          holds deployer password for cloudgov space"
+        type: env_var_name
+      cloudgov_space:
+        description: "Name of CircleCi project environment variable that
+          holds name of cloudgov space to target for application deployment"
+        type: env_var_name
+      deploy_config_file:
+        description: "Path to deployment configuration file"
+        type: string
+      session_secret:
+        description: "Name of CircleCi project environment variable that
+          holds session secret, a required application variable"
+        type: env_var_name
+      new_relic_license:
+        description: "Name of CircleCI project environment variable that
+          holds the New Relic License key, a required application variable"
+        type: env_var_name
+    steps:
+      - run:
+          name: Login with service account
+          command: |
+            cf login -a << pipeline.parameters.cg_api >> \
+              -u ${<< parameters.cloudgov_username >>} \
+              -p ${<< parameters.cloudgov_password >>} \
+              -o << pipeline.parameters.cg_org >> \
+              -s ${<< parameters.cloudgov_space >>}
+      - run:
+          name: Push Data Analytics Modeling Application with deployment vars
+          command: |
+            cf push --docker-image data-models REPO/IMAGE:TAG
+--strategy rolling \ #not sure if strategy is needed for docker
+              --vars-file << parameters.deploy_config_file >> \
+              --var AUTH_CLIENT_ID=${<< parameters.auth_client_id >>} \
+              --var AUTH_CLIENT_SECRET=${<< parameters.auth_client_secret >>} \
+              --var NEW_RELIC_LICENSE_KEY=${<< parameters.new_relic_license >>} \
+              --var SESSION_SECRET=${<< parameters.session_secret >>}
+              ...
+```
+### Node Child Processes
```

Comment:
My only thought on the two options is that child processes make me kind of uneasy, but that could be unfounded because I can't really elaborate why. As long as it is doable resource wise, it may be the easiest way to proceed.

Reply:
I'm not a fan of managing an additional set of package dependencies that may or may not persist.